qiita_db/portal.py

```diff
@@ -32,32 +34,198 @@ def get_studies_by_portal(portal):
         portal : str
             Portal to associate with.
+
+        Raises
+        ------
+        ValueError
+            Try and delete from QIITA portal
         """
         if portal == "QIITA":
             raise ValueError('Can not remove from main QIITA portal!')
+        _check_studies(studies)
+
+        conn_handler = SQLConnectionHandler()
+        # Clean list of studies down to ones associated with portal already
+        portal_id = convert_to_id(portal, 'portal_type', 'portal')
+        conn_handler = SQLConnectionHandler()
+        sql = """SELECT study_id from qiita.study_portal
+                 WHERE portal_type_id = %s AND study_id IN %s"""
+        clean_studies = [x[0] for x in conn_handler.execute_fetchall(
+                         sql, [portal_id, tuple(studies)])]
+
+        if len(clean_studies) != len(studies):
+            rem = map(str, set(studies).difference(clean_studies))
+            warnings.warn("The following studies are not part of %s: %s" %
+                          (portal, ', '.join(rem)))
+
+        sql = """DELETE FROM qiita.study_portal
+                 WHERE study_id IN %s AND portal_type_id = %s"""
+        if len(clean_studies) != 0:
+            conn_handler.execute(sql, [tuple(studies), portal_id])
+
+    def get_analyses_by_portal(portal):
+        """Returns analysis id for all Analyses belonging to a portal
+
+        Parameters
+        ----------
+        portal : str
+           Portal to check analyses belong to
+
+        Returns
+        -------
+        set of int
+            All analysis ids in the database that match the given portal
+        """
+        conn_handler = SQLConnectionHandler()
+        portal_id = convert_to_id(portal, 'portal_type', 'portal')
+        sql = """SELECT analysis_id FROM qiita.analysis_portal
+                 WHERE portal_type_id = %s"""
+        return {x[0] for x in
+                conn_handler.execute_fetchall(sql, [portal_id])}
+
+    def _check_analyses(analyses):
+        conn_handler = SQLConnectionHandler()
+        # Check if any analysis IDs given do not exist.
+        sql = "SELECT analysis_id from qiita.analysis WHERE analysis_id IN %s"
+        existing = [x[0] for x in conn_handler.execute_fetchall(
+            sql, [tuple(analyses)])]
+        if len(existing) != len(analyses):
+            bad = map(str, set(analyses).difference(existing))
+            raise QiitaDBError("The following analyses do not exist: %s" %
+                               ", ".join(bad))
+
+        # Check if any analyses given are default
+        sql = ("SELECT analysis_id from qiita.analysis WHERE analysis_id IN %s"
+               " AND dflt = True")
+        default = [x[0] for x in conn_handler.execute_fetchall(
+            sql, [tuple(analyses)])]
+        if len(default) > 0:
+            bad = map(str, set(analyses).difference(default))
+            raise QiitaDBError("The following analyses are default: %s" %
```

Comment:
Actually, this is better:
`The following analyses are default and can't be deleted or assigned to another portal: %s`